cmd/thanos/sidecar.go

```diff
@@ -125,6 +139,31 @@ func runSidecar(
 
 		ctx, cancel := context.WithCancel(context.Background())
 		g.Add(func() error {
+			// Only check Prometheus's flags when upload is enabled.
+			if uploads {
+				// Retry infinitely until we get Prometheus version.
+				if err := runutil.Retry(2*time.Second, ctx.Done(), func() error {
+					if err := m.FetchPromVersion(logger); err != nil {
+						level.Warn(logger).Log(
+							"msg", "failed to get Prometheus version. Is Prometheus running? Retrying",
+							"err", err,
+						)
+						return errors.Wrapf(err, "fetch Prometheus version")
+					}
+
+					return nil
+				}); err != nil {
+					return err
+				}
+
+				// Check prometheus's flags to ensure sane sidecar flags.
+				if err := validatePrometheus(ctx, logger, m); err != nil {
+					return errors.Wrap(err, "validate Prometheus flags")
+				}
+			} else {
+				level.Info(logger).Log("msg", "uploading compacted blocks is disabled, skip validation")
```

Comment:
We should probably change this message too accordingly :smile:

Reply:
Sorry for this low quality patch. Done.

Reply:
It's ok, no worries :smile: there is no rush and everything's ok :+1: